schemas/stsci.edu/asdf/transform/remap_axes-1.3.0.yaml

```diff
@@ -0,0 +1,94 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/transform/remap_axes-1.3.0"
+tag: "tag:stsci.edu:asdf/transform/remap_axes-1.3.0"
+title: >
+  Reorder, add and drop axes.
+
+description: |
+  This transform allows the order of the input axes to be shuffled and
+  returned as the output axes.
+
+  It is a list made up of integers or "constant markers".  Each item
+  in the list corresponds to an output axis.  For each item:
+
+  - If an integer, it is the index of the input axis to send to the
+    output axis.
+
+  - If a constant, it must be a single item which is a constant value
+    to send to the output axis.
+
+  If only a list is provided, the number of input axes is
+  automatically determined from the maximum index in the list.  If an
+  object with `mapping` and `n_inputs` properties is provided, the
+  number of input axes is explicitly set by the `n_inputs` value.
+
+  Invertibility: TBD
+examples:
+  -
+    - For 2 input axes, swap the axes
+    - |
+        !transform/remap_axes-1.3.0
+          mapping: [1, 0]
+  -
+    - For 2 input axes, return the second axis and drop the first
+    - |
+        !transform/remap_axes-1.3.0
+          mapping: [1]
+
+  -
+    - For 2 input axes, return the first axis twice, followed by the second
+    - |
+        !transform/remap_axes-1.3.0
+          mapping: [0, 0, 1]
+
+  -
+    - For 2 input axes, add a third axis which is a constant
+    - |
+        !transform/remap_axes-1.3.0
+          mapping: [0, 1, !core/constant-1.0.0 42]
+
+  -
+    - |
+        The above example is equivalent to the following, and ASDF
+        implementations are free to normalize it thusly:
+    - |
+        !transform/concatenate-1.2.0
+          forward:
+            - !transform/remap_axes-1.3.0
+              mapping: [0]
+            - !transform/remap_axes-1.3.0
+              mapping: [1]
+            - !transform/constant-1.4.0
+              value: 42
+              dimensions: 1
+
+  -
+    - Here we have 3 input axes, but we are explicitly dropping the last one
+    - |
+        !transform/remap_axes-1.3.0
+          mapping: [0, 1]
+          n_inputs: 3
+
+definitions:
+  mapping:
+    type: array
+    items:
+      anyOf:
+        - type: integer
+        - $ref: "../core/constant-1.0.0"
```

Comment:
I wonder why this is defined as a constant and not integer. If I had to write the schema I would have written the items as integers. Any ideas?

Reply:
I can't figure out why we have core/constant at all -- it's only used in remap_axes, and it doesn't seem like the RemapAxesType in astropy can ever write a core/constant:

https://github.com/astropy/astropy/blob/master/astropy/io/misc/asdf/tags/transform/compound.py#L143

And on read, it turns a non-integer node into a Const1D anyway.  Maybe core/constant is a failed experiment that just hasn't been cleaned up yet?

Reply:
Per discussion on Slack, let's remove core/constant here, since the astropy [tag class](https://github.com/astropy/astropy/blob/master/astropy/io/misc/asdf/tags/transform/compound.py#L148) no longer writes it anyway.

Reply:
should i remove core/constant from the newest version map too?

Reply:
I think we should leave it for now, the value of that schema isn't clear to me but it's been in the core schemas for a while.  We should probably seek community input before removing it.